api/src/main/java/com/vinumeris/updatefx/UpdateFX.java

```diff
@@ -85,9 +85,17 @@ public static Path findAppExecutable(Path appInstallDir) throws IOException {
         if (os.contains("mac")) {
             // Returns path to the .app directory, which normally contains only a Contents directory.
             List<Path> exes = Utils.listDir(appInstallDir.resolve("Contents/MacOS"));
-            if (exes.size() != 1)
-                throw new IllegalStateException("Found unknown number of app executables");
-            return exes.get(0);
+            for (Path exe : exes) {
+                /**
+                 * Since 8u40, the packager introduced a "libpackager.dylib"
```

Comment:
I have another version where I check if the file name has a "." (dot) inside. If not, then I take the file. Thus we avoid putting any string directly for comparison, does that sound better?

if (!exe.getFileName().toString().contains(".")) {
                    return exe;
                }